posts/2021-02-19-mpcontextpropagation-requesttiming-21003-beta.adoc

```diff
@@ -0,0 +1,314 @@
+
+The 1.2 release of MicroProfile Context Propagation aligns with the MicroProfile 4.0 platform, specifically addressing a difference in how MicroProfile Config 2.0 treats empty value configuration properties. When using MicroProfile Config to specify an empty list of thread context types for MicroProfile Context Propagation to use as defaults, use a value of `None` rather than an empty value. An empty value in MicroProfile Config 2.0 indicates to override any lower ordinal config sources and instead use the built-in default value for the property.  For example, the combination of `mp.context.ManagedExecutor.cleared=None` and `mp.context.ManagedExecutor.propagated=Remaining` causes every context type to be propagated.
+
+To enable the MicroProfile Context Propagation 1.2 feature, add the following to your server configuration,
+
+[source, xml]
+----
+<featureManager>
+  <feature>mpContextPropagation-1.2</feature>
+  <!-- other features used by example code... -->
+  <feature>servlet-4.0</feature>
+  <feature>jdbc-4.2</feature>
+  <feature>jndi-1.0</feature>
+</featureManager>
+----
+
+Example usage within a Servlet:
+
+[source, java]
+----
+private ManagedExecutor executor;
+
+public void init(ServletConfig config) throws ServletException {
+    executor = ManagedExecutor.builder()
+                .propagated(ThreadContext.APPLICATION)
+                .cleared(ThreadContext.ALL_REMAINING)
+                .build();
+}
+
+public void destroy() {
+    executor.shutdownNow();
+}
+
+public void doGet(HttpServletRequest req, HttpServletResponse resp)
+    throws ServletException, IOException {
+    ...
+    executor.copy(unmanagedCompletionStage).thenAcceptAsync(value -> {
+        // requires java:comp namespace of the application,
+        DataSource ds = InitialContext.doLookup("java:comp/env/jdbc/ds");
+        ...
+    });
+}
+----
+
+Find out more: 
+
+* link:https://download.eclipse.org/microprofile/microprofile-context-propagation-1.2-RC1/microprofile-context-propagation-spec-1.2-RC1.html[The MicroProfile Context Propagation 1.2 Release Candidate 1 specification]
+* link:https://download.eclipse.org/microprofile/microprofile-context-propagation-1.2-RC1/apidocs/[The MicroProfile Context Propagation 1.2 JavaDoc]
+
+[#leakedConnections]
+=== Automatic Cleanup of Leaked Connections
+
+Liberty connection management is enhanced with the ability to automatically detect and close unsharable connections that are left open by the application across the end of a request.
+
+Occasionally, application code might forget to close an unsharable connection that it obtains, which prevents the connection from being returned to the connection pool for use by other requests. Over time, these leaked connections can degrade performance and eventually exhaust the connection pool.  Liberty connection management now has the ability to detect and automatically close these leaked connections to prevent this from happening.
```

Comment:
```suggestion
Occasionally, application code might forget to close an unsharable connection that it obtains. This prevents the connection from being returned to the connection pool for use by other requests. Over time, these leaked connections can degrade performance and eventually exhaust the connection pool. Open Liberty connection management now has the ability to detect and automatically close these leaked connections to prevent this from happening.
```

Reply:
Done